pandas/core/generic.py

```diff
@@ -200,7 +199,7 @@ def __init__(
         self,
         data: BlockManager,
         copy: bool = False,
-        attrs: Optional[Mapping[Optional[Hashable], Any]] = None,
+        attrs: Optional[Mapping[Label, Any]] = None,
```

Comment:
I think this may be one case @WillAyd was thinking of. `attrs` needn't have any any relation with the columns of the DataFrame, so using `Label` may be confusing.

Reply:
Yea this is a great example (thanks @TomAugspurger for following up for me!)